### PR TITLE
Booking Period Implemetation

### PR DESCRIPTION
**Pull request by Benjamin-S**
Tested locally and runs correctly. Adding to pull request so that you can review it

### CODE REVIEW

File: src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs [hunk 1]
```diff
@@ -11,28 +11,46 @@ public BookingPeriod(DateTime start, DateTime end)
         {
             Start = start;
             End = end;
+
+            if (start == end)
```
Comment (tharinga) [inline src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs]: Can be combined into a single if statement. Always use braces, even when optional. Constructor assignments and logic should come after the guard clause. 
```suggestion
            if (end <= start)
            {
                throw  new ArgumentException("End cannot be before or equal to start time");
            }
            Start = start;
	    End = end;
```

File: src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs [hunk 2]
```diff
@@ -11,28 +11,46 @@ public BookingPeriod(DateTime start, DateTime end)
         {
```
Comment (tharinga) [inline src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs]: Guard clause can be combined into a single if statement. 
Always use braces, even when optional. 
Constructor assignments and logic should come after the guard clause. 

```suggestion
            if (end <= start)
            {
                throw  new ArgumentException("End cannot be before or equal to start time");
            }
            Start = start;    
            End = end;
```

File: src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs [hunk 3]
```diff
@@ -11,28 +11,46 @@ public BookingPeriod(DateTime start, DateTime end)
...
+            if(end < start)
+                throw new ArgumentException("End date precedes start date");
         }
         public static BookingPeriod CreateOneHourPeriod(DateTime day)
         {
-            throw new NotImplementedException();
+            var start = day;
+            var end = day.AddHours(1);
+
+            return(new BookingPeriod(start, end));
```
Comment (tharinga) [inline src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs]: Creation of temporary start and end variables is redundant.
Return statements do not require braces.
This suggestion also applies to the other functions.
```suggestion
            return new BookingPeriod(day, day.AddHours(1));
```

File: src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs [hunk 4]
```diff
@@ -11,28 +11,46 @@ public BookingPeriod(DateTime start, DateTime end)
...
-            throw new NotImplementedException();
+            var start = startDateTime;
+            var end = startDateTime.AddMonths(1);
+
+            return(new BookingPeriod(start, end));
         }
         public bool Overlaps(BookingPeriod dateTimeRange)
         {
-            throw new NotImplementedException();
+            return (this.Start < dateTimeRange.End && dateTimeRange.Start < this.End);
```
Comment (tharinga) [inline src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs]: Logically equivalent but slightly more readable
```suggestion
            return this.Start < dateTimeRange.End && 
                   this.End > dateTimeRange.Start;
```

**Comment by Benjamin-S** [comment]
I've made your recommended changes and have requested a new review.

**Pull request merged**